transcom/mymove
# B19795 int doc viewer doc title change

## Agility ticket
[B19795](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19795)

## Summary
For documents uploaded in the app, the documents were to have the document name in the format of <filename>-<timestamp> where timestamp is yyyymmddhh24minssecs e.g. sample.pdf-20240812134209.  In addition, when the menu item is clicked in the document viewer, document name was to wrap (instead of being truncated). 

### How to test
testing prep:
* suggest create a file w/a very long name e.g. "sample orders_company alpha_battalion twleve.pdf".  This is sufficiently long enough to test document name wrapping in the document viewer

========
* HHG move
  - customer
    * create a HHG move
    * submit move
    * go to: upload additional docs
      - upload file with long filename
      - confirm it uploaded in the format: filename-timestamp
    * go to: orders > upload documents
      - repeat same steps as "upload additional docs"
    * go to: review your request > orders > edit
      - repeat same steps as "upload additional docs"
  - service counselor
    * go to: view edit orders
      - in the doc viewer pane on the left, click the menu button
      - verify documents in the list wrap
      - click: manage orders
        * upload file
        * confirm it uploaded in the format: filename-timestamp
        * doc viewer: click the menu button and verify it appears
        * click "done"
      - click: manage amended orders
        * upload file
        * confirm it uploaded in the format: filename-timestamp
        * doc viewer: click the menu button and verify it appears
        * click "done"
      - fix order fields
    * go to: supporting docs
      - in the doc viewer pane on the left, click the menu button
      - verify documents in the list wrap
      - upload file
      - confirm it uploaded in the format: filename-timestamp
      - doc viewer: click the menu button and verify it appears
    * submit move
  - too:  approve shipment
  - prime
    * create payment request
    * upload document with long name (to verify it wraps in the viewer)
      - note: unlike other upload locations in the app, when the file is uploaded, it will not have a timestamp in the name.  the timestamp actually gets appended in the backend after "save" is clicked.
  - tio
    * click: review service items
    * doc viewer, click the menu button
      - confirm the file uploaded in the format: filename-timestamp
      - confirm it wraps in the menu list

========
* PPM move
  - customer
    * create PPM move
    * submit move
  - service counselor: submit move
  - customer
    * click "upload ppm docs"
    * vehicle weight
      - empty, full docs:  upload doc with long filename
    * add pro gear
      - customer: create pro gear item.  upload doc with long filename
      - spouse: create pro gear item.  upload doc with long filename
    * add expense: upload doc with long filename
    * submit documentation
  - service counselor
    * review docs
      - for each page
        * click the "menu" item 
          - confirm each item in the list is in the format: filename-timestamp
          - confirm the item wraps if it is too long


## ANDI Screenshots
No actionable UI elements were added

## Testing Screenshots
none
